language-plutus-core/src/Language/PlutusCore/Error.hs

```diff
@@ -42,11 +44,18 @@ data RenameError a
     | UnboundTyVar (TyName a)
     deriving (Show, Eq, Generic, NFData)
 
+-- | This error is returned whenever scope resolution of a 'DynamicBuiltinName' fails.
+newtype UnknownDynamicBuiltinNameError
+    = UnknownDynamicBuiltinNameError DynamicBuiltinName
+    deriving (Show, Eq, Generic)
+    deriving newtype (NFData)
+
 data TypeError a
     = KindMismatch a (Type TyNameWithKind ()) (Kind ()) (Kind ())
     | TypeMismatch a (Term TyNameWithKind NameWithType ())
                      (Type TyNameWithKind ())
                      (NormalizedType TyNameWithKind ())
+    | UnknownDynamicBuiltinNameTypeError UnknownDynamicBuiltinNameError
```

Comment:
Couldn't this just be `UnknownDynamicBuiltinError DynamicBuiltinName` or the like? This seems a little heavy.

Reply:
Yes, but I specifically made a separate `UnknownDynamicBuiltinNameError` type, because other procedures may return such errors as well. E.g. the parser, or evaluators (which currently don't). I agree it's heavy.

Reply:
> perhaps we can save a little length by dropping `Name`. `UnknownDynamicBuiltin` seems descriptive enough! Or even `UnknownDynBuiltin`.

Exactly what I want to do later. The task was to add dynamic built-in names, but I want to argue for adding things that are not only names.

`DynBuiltinName` is what I started with, but there is a super annoying problem: how to extend

```haskell
data Constant a = BuiltinInt a Natural Integer
                | BuiltinBS a Natural BSL.ByteString
                | BuiltinSize a Natural
                | BuiltinName a BuiltinName
```

? I added

```haskell
                | DynBuiltinName a DynamicBuiltinName
```

there.

Now I think that it probably should be

```haskell
                |  DynamicBuiltinName a DynBuiltinName
```

And I can drop the `Name` suffix as well.

So if people prefer, I can do the renaming in this PR. Suggestions on how to extend the `Constant` type with a dynamic built-ins constructor are highly appreciated.

Reply:
I didn't mean the more general issue of other kinds of builtin, I just meant we could drop it from the name in the error ;)